xcode/Subconscious/Shared/Services/DatabaseService.swift

```diff
@@ -291,45 +291,58 @@ final class DatabaseService {
             )
         })
     }
-
-    /// Purge all content of sphere with given DID from the database.
+ 
+    /// Purge all content of sphere with given petname from the database.
+    /// This method accounts for the possibility of following the same DID
+    /// under multiple names.
     ///
     /// This does not delete the content from file system or sphere,
     /// only removes knowledge of it from the database.
-    func purgePeer(identity: Did) throws {
+    func purgePeer(petname: Petname) throws {
         guard self.state == .ready else {
             throw DatabaseServiceError.notReady
         }
         let savepoint = "purge"
+        
+        guard let peer = try readPeer(petname: petname) else {
```

Comment:
Is this guard clause necessary? If we delete a peer, but there are no peers with that name, the end result should be the same.

Reply:
We need to know the DID of a peer to delete from the `memos` table, which we can only get by reading the peer.